### ISSUE TITLE
Support tagged digested references

### ISSUE THREAD

**Issue by glyn** [camallo/dkregistry-rs#144]
`dkregistry::reference::Version` does not accommodate tagged digested references. Although these are not particularly common, they are valid.

**Comment by lucab** [comment]
Indeed. Related, it would probably be a good idea to first address this TODO item for parsing references: https://github.com/camallo/dkregistry-rs/blob/cb22b538d788f7a88bf390e34bece4a503cefecb/src/reference.rs#L156

**Comment by glyn** [comment]
Perhaps. One starting point for a reference grammar is in a [code comment](https://github.com/docker/distribution/blob/a8371794149d1d95f1e846744b05c87f2f825e5a/reference/reference.go#L6); I am not aware of a grammar in the OCI specs. The [testcases](https://github.com/docker/distribution/blob/a8371794149d1d95f1e846744b05c87f2f825e5a/reference/reference_test.go) clarify the intention though.
